x/currency/model.go

```diff
@@ -0,0 +1,61 @@
+package currency
+
+import (
+	"regexp"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+var _ orm.CloneableData = (*TokenInfo)(nil)
+
+func (t *TokenInfo) Validate() error {
+	if !isTokenName(t.Name) {
+		return ErrInvalidTokenName(t.Name)
+	}
+	if t.SigFigs < minSigFigs || t.SigFigs > maxSigFigs {
+		return ErrInvalidSigFigs(t.SigFigs)
+	}
+	return nil
+}
+
+var isTokenName = regexp.MustCompile(`^[A-Za-z0-9 \-_:]{3,32}$`).MatchString
+
+const (
```

Comment:
req: const on top
req: vars below const

Reply:
I would like to challenge this.

There are many styles and conventions. Some languages like Python, are having rules to structure file module import first, then constant and global variable declarations, classes and finally functions.

It is encouraged in Go and I think it is better style to [write code top down](https://talks.golang.org/2014/readability.slide#4). Why force the reader to jump around the file when reading? Instead write it the way, that reading comes natural.

I understand that other places are using Python (or other community) like format, but maybe it's work rethinking this choice?

Reply:
I would really love to have constants and vars up top too, along with vars below them
It's how we do it everywhere in our code currently and the whole concept of having consts right before they are being used has it's dark side, when you need to see all the consts or add some more usages. In the end it could end up being a mess, whereas with the default approach you always know where to find what. 

While this could be a nice optimisation it's not worth it most of the time. And yes, let's be consistent.

Reply:
Let's be consistent in this project and try something new in the next one.